include/networkit/scd/SelectiveCommunityDetector.hpp

```diff
@@ -16,22 +16,57 @@
 
 namespace NetworKit {
 
+/**
+ * Base class for selective community detection algorithms, i.e., algorithms
+ * that detect communities around a single node.
+ */
 class SelectiveCommunityDetector {
 
 public:
+    /**
+     * Construct a selective community detector. This should only be called by child classes.
+     *
+     * @param G The graph for which communities shall be detected.
+     */
+    SelectiveCommunityDetector(const Graph& G);
 
     /**
-     * @param[in] G the input graph.
+     * Virtual default destructor to allow safe destruction of child classes.
      */
-    SelectiveCommunityDetector(const Graph& G) : G(&G) {}
     virtual ~SelectiveCommunityDetector() = default;
 
     /**
-     * Detect communities for given seed nodes.
+     * Detect one community for each of the given seed nodes.
+     *
+     * The default implementation calls expandOneCommunity() for each of the seeds.
+     *
+     * @param seeds The list of seeds for which communities shall be detected.
      * @return a mapping from seed node to community (as a set of nodes)
      */
-    virtual std::map<node, std::set<node>> run(const std::set<node>& seeds) = 0;
+    virtual std::map<node, std::set<node> >  run(const std::set<node>& seeds);
+
+    /**
+     * Detect a community for the given seed node.
+     *
+     * The default implementation calls expandOneCommunity(const
+     * std::set<node>&) with a set of one node.
+     *
+     * @param seed The seed to find the community for.
+     * @return The found community as set of node.
+     */
+    virtual std::set<node> expandOneCommunity(node seed);
 
+    /**
+     * Detect a single community for the given seed nodes.
+     *
+     * This is useful if you know multiple nodes that should be part of the
+     * community. This method may throw an exception if the particular algorithm
+     * does not support multiple seeds.
+     *
+     * @param seeds The seeds for the community.
+     * @return The found community as set of nodes.
+     */
+    virtual std::set<node> expandOneCommunity(const std::set<node>& seeds) = 0;
```

Comment:
Instead of a `std::set`, what about a range of nodes?

Reply:
Accepting a range of nodes means that this method must be a template method, but template methods cannot be virtual. Having this method as a virtual method allows accepting any selective community detector and executing it from a set of seed nodes.

Reply:
Ah that's right. In the implementation of this method the order in which you process the seeds does not seem to be relevant. Would that be better to use a `std::unordered_set` for `seeds`?

Reply:
I wanted to keep this consistent with the return type of the already existing local community detection algorithms, which is `std::set`, as my primary application of multiple seed nodes is to use the result of a local community detection algorithm (in particular `CliqueDetect`) as seed for another local community detection algorithm. I would also be open for a change to `std::vector` as it should have much less overhead compared to `std::set` or `std::unordered_set`.